tasks/main.yml

```diff
@@ -10,47 +11,83 @@
     - intel-network-adapter
     - mellanox-network-adapter
 
-- include_tasks: ilo.yml
+- name: Run tasks for ILO4 firmware (gen8 and gen9)
+  include_tasks: ilo.yml
   tags:
     - ilo
-  when: upgrade_ilo_firmware|bool
+  when: |
+    upgrade_ilo_firmware and
+    (hp_gen == "gen8") or (hp_gen == "gen9")
 
-- include_tasks: system_rom.yml
+- name: Run tasks for system ROM firmware (gen8 and gen9)
+  include_tasks: system_rom.yml
   tags:
    - system-rom
-  when: upgrade_system_rom_firmware|bool
+  when: |
+    upgrade_system_rom_firmware and
+    (hp_gen == "gen8") or (hp_gen == "gen9")
 
-- include_tasks: disk_controller.yml
+- name: Run tasks for disk controller firmware (gen8 and gen9)
+  include_tasks: disk_controller.yml
   tags:
    - disk-controller
-  when: upgrade_disk_controller_firmware|bool
+  when: |
+    upgrade_disk_controller_firmware and
+    (hp_gen == "gen8") or (hp_gen == "gen9")
 
-- include_tasks: disk_drive.yml
+- name: Run tasks for disk drive firmware (gen8 and gen9)
+  include_tasks: disk_drive.yml
   tags:
    - disk-drive
-  when: upgrade_disk_drive_firmware|bool
+  when: |
+    upgrade_disk_drive_firmware and
+    (hp_gen == "gen8") or (hp_gen == "gen9")
 
-- include_tasks: disk_backplane_expander.yml
+- name: Run tasks for backplane expander firmware (gen8 and gen9)
+  include_tasks: disk_backplane_expander.yml
   tags:
    - disk-backplane-expander
-  when: upgrade_disk_backplane_expander|bool
+  when: |
+    upgrade_disk_controller_firmware and
```

Comment:
is there any reason for using the `upgrade_disk_controller_firmware` flag instead of the `upgrade_disk_backplane_expander` flag in this specific task? Same applies for the tasks below

Reply:
No, it's a typo. Thanks for pointing out. Fixing soon.